codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/HttpBoundProtocolGenerator.kt

```diff
@@ -383,11 +383,14 @@ class HttpBoundProtocolGenerator(
         httpBindingGenerator.renderUpdateHttpBuilder(implBlockWriter)
         httpBuilderFun(implBlockWriter) {
             rust("let builder = #T::new();", RuntimeType.HttpRequestBuilder)
+            rust("let mut builder = self.update_http_builder(builder)?;")
             val additionalHeaders = listOf("Content-Type" to contentType) + protocol.additionalHeaders(operationShape)
             for (header in additionalHeaders) {
-                rust("let builder = builder.header(${header.first.dq()}, ${header.second.dq()});")
+                withBlock("if !builder.headers_ref().map(|headers| headers.contains_key(${header.first.dq()})).unwrap_or(false) {", "}") {
+                    rust("builder = builder.header(${header.first.dq()}, ${header.second.dq()});")
+                }
```

Comment:
small simplification
```suggestion
                rustTemplate("""
                if !builder.headers_ref().unwrap_or(&#{http}::HeaderMap::new()).contains_key(${header.first.dq()}) {
                    builder = builder.header(${header.first.dq()}, ${header.second.dq()});
                }""", "http" to RuntimeType.http)
```